### PR TITLE
chore: Replace inbox mixin with inbox helper [CW-3468]

### PR DESCRIPTION
**Pull request by fayazara**
# Pull Request Template

## Description

Replaced inbox mixin with a new helper method

### CODE REVIEW

File: app/javascript/shared/helpers/inboxHelper.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  [INBOX_TYPES.TELEGRAM]: inbox => inbox.channel_type === INBOX_TYPES.TELEGRAM,
+  [INBOX_TYPES.SMS]: inbox => inbox.channel_type === INBOX_TYPES.SMS,
+  [INBOX_TYPES.WHATSAPP]: inbox => inbox.channel_type === INBOX_TYPES.WHATSAPP,
+};
+
+export const getChannelType = inbox => inbox.channel_type;
+
+export const getWhatsAppAPIProvider = inbox => inbox.provider || '';
+
+export const isChannelType = (inbox, type) => channelsMap[type](inbox);
```
Comment (muhsin-k) [inline app/javascript/shared/helpers/inboxHelper.js:20]:
```suggestion
export const isChannelType = (inbox, type) =>
  channelTypeCheckers[type]?.(inbox) || false;
```

File: app/javascript/shared/helpers/inboxHelper.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+export const isAWhatsAppCloudChannel = inbox =>
+  isChannelType(inbox, INBOX_TYPES.WHATSAPP) &&
+  getWhatsAppAPIProvider(inbox) === 'whatsapp_cloud';
+
+export const is360DialogWhatsAppChannel = inbox => {
+  return (
+    inbox.channel_type === INBOX_TYPES.WHATSAPP &&
+    inbox.provider === '360dialog'
+  );
+};
```
Comment (muhsin-k) [inline app/javascript/shared/helpers/inboxHelper.js:65]:
```suggestion
export const is360DialogWhatsAppChannel = inbox =>
  inbox.channel_type === INBOX_TYPES.WHATSAPP && inbox.provider === '360dialog';
```

File: app/javascript/shared/helpers/inboxHelper.js (hunk 1 above)
Comment (muhsin-k) [reply app/javascript/shared/helpers/inboxHelper.js:20]: Lets provide a default false value.

**Review by muhsin-k — APPROVED** [review]
The code looks good to me. Lets test this properly before merging.

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 3]
```diff
@@ -129,16 +148,16 @@ export default {
...
+        !is360DialogWhatsAppChannel(this.inbox)
       );
     },
     showRichContentEditor() {
       if (this.isOnPrivateNote || this.isRichEditorEnabled) {
         return true;
       }
 
-      if (this.isAPIInbox) {
+      if (this.isAPIInbox(this.inbox)) {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:160]: Let use directly from helper itself
```suggestion
      if (isAPIInbox(this.inbox)) {
```

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 4]
```diff
@@ -244,14 +263,14 @@ export default {
...
-        this.isAWhatsAppChannel ||
-        this.isAPIInbox ||
-        this.isAnEmailChannel ||
-        this.isASmsInbox ||
-        this.isATelegramChannel ||
-        this.isALineChannel
+        isAWebWidgetInbox(this.inbox) ||
+        isAFacebookInbox(this.inbox) ||
+        isAWhatsAppChannel(this.inbox) ||
+        this.isAPIInbox(this.inbox) ||
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:269]:
```suggestion
        isAPIInbox(this.inbox) ||
```

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 5]
```diff
@@ -315,14 +334,14 @@ export default {
...
     },
     enableMultipleFileUpload() {
       return (
-        this.isAnEmailChannel ||
-        this.isAWebWidgetInbox ||
-        this.isAPIInbox ||
-        this.isAWhatsAppChannel
+        isAnEmailChannel(this.inbox) ||
+        isAWebWidgetInbox(this.inbox) ||
+        this.isAPIInbox(this.inbox) ||
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:343]:
```suggestion
        isAPIInbox(this.inbox) ||
```

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 6]
```diff
@@ -354,10 +373,10 @@ export default {
       return `draft-${this.conversationIdByRoute}-${this.replyType}`;
     },
     audioRecordFormat() {
-      if (this.isAWhatsAppChannel || this.isATelegramChannel) {
+      if (isAWhatsAppChannel(this.inbox) || isATelegramChannel(this.inbox)) {
         return AUDIO_FORMATS.MP3;
       }
-      if (this.isAPIInbox) {
+      if (this.isAPIInbox(this.inbox)) {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:379]:
```suggestion
      if (isAPIInbox(this.inbox)) {
```

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 7]
```diff
@@ -115,13 +130,13 @@ export default {
       }
 
       if (
-        this.isATwilioChannel ||
-        this.isALineChannel ||
+        isATwilioChannel(this.inbox) ||
+        isALineChannel(this.inbox) ||
         this.isAPIInbox ||
         (this.isAnEmailChannel && !this.inbox.provider) ||
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:136]: @fayazara Does this `this.isAPIInbox` and `this.isAnEmailChannel` works? We need to pass the inbox to the helper, right?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 8]
```diff
@@ -115,13 +130,13 @@ export default {
...
-        this.isAMicrosoftInbox ||
-        this.isAGoogleInbox ||
-        this.isAWhatsAppChannel ||
+        isAMicrosoftInbox(this.inbox) ||
+        isAGoogleInbox(this.inbox) ||
+        isAWhatsAppChannel(this.inbox) ||
         this.isAWebWidgetInbox
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:140]: Does this `this.isAWebWidgetInbox` work?

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 9]
```diff
@@ -158,12 +173,15 @@ export default {
...
+      ) {
         return `${this.inbox.name} (${
           this.inbox.messaging_service_sid || this.inbox.phone_number
         })`;
       }
-      if (this.isAWhatsAppChannel) {
+      if (isAWhatsAppChannel(this.inbox)) {
         return `${this.inbox.name} (${this.inbox.phone_number})`;
       }
       if (this.isAnEmailChannel) {
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:187]: Does this works? `this.isAnEmailChannel`

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 10]
```diff
@@ -173,7 +191,9 @@ export default {
     },
     canLocktoSingleConversation() {
       return (
-        this.isASmsInbox || this.isAWhatsAppChannel || this.isAFacebookInbox
+        isASmsInbox(this.inbox) ||
+        isAWhatsAppChannel(this.inbox) ||
+        isAFacebookInbox(this.inbox)
       );
     },
     inboxNameLabel() {
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:199]: In this computed property, we are not using the `isAWebWidgetInbox` helper. 
And same for this `inboxNamePlaceHolder` computed property.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:1]: @fayazara Can you cross-check this component? Still, we are directly using some of the helpers without passing the inbox data.

File: app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue [hunk 11]
```diff
@@ -228,13 +247,13 @@ export default {
...
       }
-      if (this.isAWhatsAppChannel) {
+      if (isAWhatsAppChannel(this.inbox)) {
         return MESSAGE_MAX_LENGTH.TWILIO_WHATSAPP;
       }
-      if (this.isASmsInbox) {
+      if (isASmsInbox(this.inbox)) {
         return MESSAGE_MAX_LENGTH.TWILIO_SMS;
       }
       if (this.isAnEmailChannel) {
```
Comment (iamsivin) [inline app/javascript/dashboard/components/widgets/conversation/ReplyBox.vue:259]: Please take a look at this `this.isAnEmailChannel`.

File: app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue [hunk 12]
```diff
@@ -138,7 +153,7 @@ export default {
           this.accountId,
           FEATURE_FLAGS.AGENT_BOTS
         ) &&
-        !(this.isAnEmailChannel || this.isATwitterInbox)
+        !(this.isAnEmailChannel || isATwitterInbox(this.inbox))
```
Comment (iamsivin) [inline app/javascript/dashboard/routes/dashboard/settings/inbox/Settings.vue:156]: See `this.isAnEmailChannel`

**Review by iamsivin — COMMENTED** [review]
@fayazara Added some comments. Please take a look. 

And the can you take a look at these files, we are not using the inbox helper there before it was handled with mixin?
1. app/javascript/dashboard/routes/dashboard/settings/inbox/channels/Website.vue

**Pull request closed without merging**